main/integration-tests/__test__/api-tests/user/update-user.test.js

```diff
@@ -0,0 +1,103 @@
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Update current user scenarios', () => {
+  let setup;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('updating current user', () => {
+    it('should fail for anonymous user', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+      await expect(anonymousSession.resources.currentUser.update({ rev: 0 })).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should update current user regardless of the uid in request body', async () => {
+      const researcher1Session = await setup.createResearcherSession();
+      const researcher2Session = await setup.createResearcherSession();
+      const researcher2Info = await researcher2Session.resources.currentUser.get();
+      const researcher1Info = await researcher1Session.resources.currentUser.get();
+      await expect(
+        researcher1Session.resources.currentUser.update({ uid: researcher2Info.uid, rev: 0, status: 'pending' }),
+      ).resolves.toEqual(expect.objectContaining({ uid: researcher1Info.uid }));
+    });
+
+    it('should fail if non-admin user update restrictive fields', async () => {
+      const researcherSession = await setup.createResearcherSession();
+      await expect(
+        researcherSession.resources.currentUser.update({ rev: 0, isSamlAuthenticatedUser: false }),
+      ).rejects.toMatchObject({ code: errorCode.http.code.forbidden });
+    });
+
+    it('should not allow alleviate privilege', async () => {
+      // Researcher should not alleviate to Admin
+      const researcherSession = await setup.createResearcherSession();
+      await expect(researcherSession.resources.currentUser.update({ rev: 0, isAdmin: true })).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+      await expect(researcherSession.resources.currentUser.update({ rev: 0, userRole: 'admin' })).rejects.toMatchObject(
+        {
+          code: errorCode.http.code.forbidden,
+        },
+      );
+      // Admin should not alleviate to root
+      const admin2Session = await setup.createAdminSession();
+      await expect(admin2Session.resources.currentUser.update({ rev: 0, userRole: 'root' })).rejects.toMatchObject({
+        code: errorCode.http.code.notFound,
+      });
+    });
+
+    it('should not allow inactive user to become active', async () => {
```

Comment:
Suggesting a change in description to something like: "'should not allow inactive user to make themselves active". Also add a test to ensure only admins can perform this operation successfully.

Reply:
The `/user` path can only be used to update current user, Admin update other user's status will be covered as part of the `/users` path.